representation/knob-representation.metta

```diff
@@ -49,3 +49,27 @@
 ;; Return: The extracted Tree
 (: getTree LogicalSubtreeKnob (Tree $a))
 (= (getTree (mkLSK (mkDiscKnob (mkKnob $tree $nodeId) $multip $curr $def $dis) $sub)) $tree)
+
+;; Calculates the knob Specification of an LSK
+(: getKnobSpec (-> LogicalSubtreeKnob DiscSpec))
+(= (getKnobSpec (mkLSK (mkDiscKnob $knob (mkMultip $multip) (mkDiscSpec $curr) (mkDiscSpec $def) $dis) $subtree))
+(mkDiscSpec (- $multip (List.length $dis))))
+
+;; A comparison function for discSpec 
+;; (used as a $compFunc to insert knobSpec and knob pairs into a multimap ) 
+(: discSpec<= (-> DiscSpec DiscSpec Bool))
+(= (discSpec<= (mkDiscSpec $newKey) (mkDiscSpec $currKey))
+(if (<= $newKey $currKey) True False))
+
+;; Converts a tuple of pairs into ordered multimap
+;; Ex: ((k1 v1) (k2 v2)) -> (ConsMMap (k1 v1) (ConsMMap (k2 v2) NilMMap))
+(: tupleToMMap (-> Expression (MultiMap ($k $v)) (-> $k $k Bool) (MultiMap ($k $v))))
```

Comment:
This function belongs in the general-helpers file and also would be great if it's name it changed to `exprToMMap` to be consistent. "We also have `exprToList` defined already"